assets/js/tournament_seat_canvas.js

```diff
@@ -0,0 +1,312 @@
+    }
+
+    createSlotOption(slot) {
+      let optionElem = document.createElement("option");
+      optionElem.value = slot;
+      optionElem.text = this.getSlotText(slot);
+      optionElem.style.color = this.getPickedColor(slot);
+      return optionElem;
+    }
+
+    redrawSlotSelection() {
+      for (let slot of Object.keys(this.slots)) {
+        let optionElem = this.slotSelection.options[this.selectionIndexes[slot]];
+        optionElem.text = this.getSlotText(slot);
+      }
+      this.validateSlots();
+    }
+
+    validateSlots() {
+      let errors = [];
+      for (let slot of Object.keys(this.slots)) {
+        if (this.slots[slot].length !== this.seatsPerSlot) {
+          errors.push("Slot " + slot.toString() + " has " + this.slots[slot].length + " seats, expected " + this.seatsPerSlot);
+        }
+      }
+      if (errors.length > 0) {
+        this.errorElem.textContent = errors.join("\r\n");
+      } else {
+        this.errorElem.textContent = "";
+      }
+    }
+
+    redrawCanvas() {
+      super.redrawCanvas();
+      this.redrawSlots();
+      this.redrawSlotSelection();
+    }
+
+    initSlotSelection() {
+      for (let [index, slot] of Object.keys(this.slots).entries()) {
+        this.slotSelection.add(this.createSlotOption(slot));
+        this.selectionIndexes[slot] = index;
+      }
+      this.slotSelection.addEventListener("change", event => {
+        this.currentSlot = Number(event.target.value);
+      });
+    }
+
+    handleAddSlotButton() {
+      // guess an unused slot number, lol
+      // django discards it later so it's fine
+      let newSlot = Math.max(...Object.keys(this.slots).map(Number), 0) + 1;
+      this.slots[newSlot] = [];
+      this.selectionIndexes[newSlot] = this.slotSelection.options.length;
+      this.slotSelection.add(this.createSlotOption(newSlot));
+      this.redrawSlotSelection();
+      this.currentSlot = newSlot;
+    }
+
+    handleRemoveSlotButton() {
+      let slot = this.currentSlot;
+      if (slot === null) {
+        return;
+      }
+      let idx = this.selectionIndexes[slot];
+      this.slotSelection.remove(idx);
+      delete this.slots[slot];
+      delete this.slotColors[slot];
+
+      // subtract 1 from indexes greater than the removed one
+      for (let [slot, index] of Object.entries(this.selectionIndexes)) {
+        if (index > idx) {
+          this.selectionIndexes[slot] -= 1;
+        }
+      }
+
+      this.currentSlot = null;
+      this.redrawCanvas();
+    }
+
+    init(seatSlots) {
+      super.init();
+      this.slots = seatSlots;
+      this.initSlotSelection();
+      this.redrawSlots();
+
+      document.getElementById("id_slot_selection_create").addEventListener("click", this.handleAddSlotButton.bind(this));
+      document.getElementById("id_slot_selection_delete").addEventListener("click", this.handleRemoveSlotButton.bind(this));
+    }
+  }
+
+
+
+  window.addEventListener('load', () => {
```

Comment:
You need to add (maybe in another grey colour) the seats locked by other tournament of the same event so that you can't pick them up.